File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/abstract_factory/IUser.java
package jv.com.tencent.cain.design_pattern.abstract_factory;

/**
 * @author : jiangyu
 * @date : 2020/3/20
 * @desc : xxx
 */
public interface IUser {
    /**
     * 插入数据
     *
     * @param user
     */
    void insertUser(User user);

    /**
     * 查询数据
     *
     * @param id
     * @return
     */
    User queryUser(int id);
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/abstract_factory/IUser.kt
package kt.com.tencent.cain.desigin_pattern.abstract_factory

/**
 * @author : jiangyu
 * @date   : 2020/3/20
 * @desc   : xxx
 */
interface IUser {
    /**
     * 插入数据
     */
    fun insertUser(user: User)

    /**
     * 查询数据
     */
    fun queryUser(id: Int): User
}